### PR TITLE
DAOS-8691 doc: Add documentation for dfuse permissions.

### PR DESCRIPTION
**Pull request by ashleypittman**
Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

File: docs/user/filesystem.md [hunk 1]
```diff
@@ -376,6 +376,32 @@ These are two command line options to control the DFuse process itself.
 These will affect all containers accessed via DFuse, regardless of any
 container attributes.
 
+### Permissions
+
+DFuse can serve data from any users container, but needs appropriate permissions in order to do
```
Comment (daltonbohning) [inline docs/user/filesystem.md:381]:
```suggestion
DFuse can serve data from any user's container, but needs appropriate permissions in order to do
```

File: docs/user/filesystem.md [hunk 2]
```diff
@@ -376,6 +376,32 @@ These are two command line options to control the DFuse process itself.
...
+this.
+
+File ownership within containers is set by the container being served, with the owner of the
+container owning all files with that container, so if looking at the container of another user
```
Comment (daltonbohning) [inline docs/user/filesystem.md:385]:
```suggestion
container owning all files within that container, so if looking at the container of another user
```

File: docs/user/filesystem.md [hunk 3]
```diff
@@ -376,6 +376,32 @@ These are two command line options to control the DFuse process itself.
...
+then all entries within that container will be owned by that user, and file based permissions
```
Comment (daltonbohning) [inline docs/user/filesystem.md:386]: nuance
```suggestion
then all entries within that container will be owned by that user, and file-based permissions
```

File: docs/user/filesystem.md [hunk 4]
```diff
@@ -376,6 +376,32 @@ These are two command line options to control the DFuse process itself.
...
+restart DFuse for these to be picked up.
+
+#### Pool permissions.
+
+DFuse needs 'r' permission for pools only.
+
+#### Container permissions.
+
+DFuse needs 'r', 't' and 'a' permissions to run, read for accessing the data, t to read container
+properties to know the container type and a to read the ACLs to know the container owner.
```
Comment (daltonbohning) [inline docs/user/filesystem.md:401]: Personally think an "oxford comma" is more clear
```suggestion
DFuse needs 'r', 't', and 'a' permissions to run: read for accessing the data, 't' to read container
properties to know the container type, and 'a' to read the ACLs to know the container owner.
```

File: docs/user/filesystem.md [hunk 5]
```diff
@@ -376,6 +376,32 @@ These are two command line options to control the DFuse process itself.
...
+
+Write permission for the container is optional, however without it the container will be read-only.
```
Comment (daltonbohning) [inline docs/user/filesystem.md:403]: nuance
```suggestion
Write permission for the container is optional; however, without it the container will be read-only.
```

**Review by daltonbohning — COMMENTED** [review]
Content looks okay - mostly just some nuanced punctuation.

**Review by daltonbohning — APPROVED** [review]

**Comment by ashleypittman** [comment]
Sending this for review now the code changes behind it have landed.

File: docs/user/filesystem.md [hunk 6]
```diff
@@ -376,6 +376,32 @@ These are two command line options to control the DFuse process itself.
...
+checks by the kernel will be made on that basis.
+
+Should write permission be granted to another user then any newly created files will also be
+owned by the container owner, regardless of the user used to create them.  Permissions are only
+checked on connect, so if permissions are revoked users need to
+restart DFuse for these to be picked up.
+
+#### Pool permissions.
+
+DFuse needs 'r' permission for pools only.
```
Comment (bfaccini) [inline docs/user/filesystem.md:396]: I am not sure I get this... Is it a real limitation or do we enforce it ??

Comment (ashleypittman) [reply docs/user/filesystem.md:396]: It will fail if you don't have r permission.

Comment (ashleypittman) [reply docs/user/filesystem.md:396]: This might be better working?
```suggestion
DFuse only needs 'r' permission for pools.
```

**Review by dsikich — APPROVED** [review]
Makes sense to me.

**Pull request merged**